File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        
        // std::string res ("");
        // int i = 0, j = 0;
        // while (i < word1.length()){
        //     res = res + word1[i];
        //     i ++;
        //     while (j < word2.length()){
        //         res = res + word2[j];
        //         j ++;
        //         break;
        //     }
        // }
        // if (j < word2.length()){
        //     res = res + word2.substr(j, word2.length());
        // }
        // return res;

        for (int i = 0; i < word2.length(); i++){
            word1.insert(2*i+1 > word1.length()? word1.length():2*i+1, word2, i, 1);
        }
        
        return word1;

    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        # res = ''
        # len1 = len(word1)
        # len2 = len(word2)
        # minlen = min(len1, len2)

        # for i in range(minlen):
        #     res += word1[i]
        #     res += word2[i]
        # if len1 > len2:
        #     res += word1[i+1:]
        # elif len2 > len1:
        #     res += word2[i+1:]

        # return res

        # check which one is shorter
        res = ''
        i, j = 0, 0
        while i < len(word1):
            res += word1[i]
            i += 1
            while j < len(word2):
                res += word2[j]
                j += 1
                break
        if j < len(word2):
            res += word2[j:]

        return res